src/liballoc/collections/linked_list.rs

```diff
@@ -878,6 +878,55 @@ impl<T> LinkedList<T> {
         unsafe { self.split_off_after_node(split_node, at) }
     }
 
+    /// Removes the element at the given index and returns it.
+    ///
+    /// # Panics
+    /// Panics if at >= len
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(linked_list_remove)]
+    /// use std::collections::LinkedList;
+    ///
+    /// let mut d = LinkedList::new();
+    ///
+    /// d.push_front(1);
+    /// d.push_front(2);
+    /// d.push_front(3);
+    ///
+    /// assert_eq!(d.remove(1), Some(2));
+    /// assert_eq!(d.remove(0), Some(3));
+    /// assert_eq!(d.remove(0), Some(1));
+    /// ```
+    #[unstable(feature = "linked_list_remove", issue = "none")]
+    pub fn remove(&mut self, at: usize) -> Option<T> {
+        let len = self.len();
+        assert!(at < len, "Cannot remove at a nonexistent index");
+
+        if at == 0 {
+            return self.pop_front();
+        } else if at == len - 1 {
+            return self.pop_back();
+        }
+
+        // Below, we iterate towards the ith node, either from the start or the end,
+        // depending on which would be faster.
+        if at <= len - at - 1 {
+            let mut cursor = self.cursor_front_mut();
+            for _ in 0..at {
+                cursor.move_next();
+            }
+            cursor.remove_current()
+        } else {
+            let mut cursor = self.cursor_back_mut();
+            for _ in 0..len - at - 1 {
```

Comment:
Can we pull out `len - at - 1` to e.g. "offset_from_end" or something like that?